sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs

```diff
@@ -242,21 +241,21 @@ public override void Dispose()
 
             readonly static HttpMethod s_patch = new HttpMethod("PATCH");
 
-            public static HttpMethod ToHttpClientMethod(HttpPipelineMethod method)
+            public static HttpMethod ToHttpClientMethod(RequestMethod method)
             {
-                switch (method)
+                switch (method.Method)
```

Comment:
Would we want similar length based checks like in `Parse` for efficiency?

Reply:
Premature optimization is biting me :) I'll copy the implementation.